cache/demux.go

```diff
@@ -99,20 +95,21 @@ func (d *demux) Unregister(w *watcher) {
 	w.Stop()
 }
 
-func (d *demux) Broadcast(event *clientv3.Event) {
+func (d *demux) Broadcast(eventBatch []*clientv3.Event) {
 	d.mu.Lock()
 	defer d.mu.Unlock()
 
-	d.history.Append(event)
+	rev := eventBatch[0].Kv.ModRevision
```

Comment:
Again events in `eventsBatch` might have different revision value. This code will not work with it.

Reply:
The current tests would not catch it because existing tests setup embedded etcd that will send events one by one if client is able to keep up. This shows that we need tests with more control over inputs.

We should move the existing cache that setup embedded etcd to `/tests/integration/` and create a new sets of tests that are based on mocked `Watcher`.